packages/flutter_tools/lib/src/run.dart

```diff
@@ -288,9 +293,78 @@ class RunAndStayResident {
     });
   }
 
+  DevFS devFS;
+
+  Future<Null> _updateDevFS() async {
+    if (devFS == null) {
+      devFS = new DevFS(Directory.current, observatory);
+
+      try {
+        await devFS.init();
+      } catch (error) {
+        devFS = null;
+        printError('Error initing DevFS: $error');
+        return null;
+      }
+    }
+
+    // Send the root and lib directories.
+    Directory directory = Directory.current;
+    _sendFiles(directory, '', _dartFiles(directory.listSync()));
+
+    directory = new Directory('lib');
+    _sendFiles(directory, 'lib', _dartFiles(directory.listSync(recursive: true, followLinks: false)));
+
+    // Send the packages.
+    if (FileSystemEntity.isFileSync(kPackagesFileName)) {
+      PackageMap packageMap = new PackageMap(kPackagesFileName);
+
+      for (String packageName in packageMap.map.keys) {
+        Uri uri = packageMap.map[packageName];
+        // Ignore self-references.
+        if (uri.toString() == 'lib/')
+          continue;
+        Directory directory = new Directory.fromUri(uri);
+        if (directory.existsSync()) {
+          _sendFiles(
+            directory,
+            'packages/$packageName',
+            _dartFiles(directory.listSync(recursive: true, followLinks: false))
+          );
+        }
+      }
+    }
+
+    try {
+      await devFS.flush();
+    } catch (error) {
+      printError('Error sending DevFS files: $error');
```

Comment:
Would users understand what "DevFS" means? That is, if this is going to be user-facing feature.